crates/chia-consensus/src/spendbundle_validation.rs

```diff
@@ -300,6 +395,65 @@ ff01\
             TEST_CONSTANTS.hard_fork_height + 1,
         )
         .expect("SpendBundle should be valid for this test");
+
+        // test wrong message
+        let solution = hex!("ffff30ffb0997cc43ed8788f841fcf3071f6f212b89ba494b6ebaf1bda88c3f9de9d968a61f3b7284a5ee13889399ca71a026549a2ff8568656c6c6f8080").to_vec();
+        // ((48 0x997cc43ed8788f841fcf3071f6f212b89ba494b6ebaf1bda88c3f9de9d968a61f3b7284a5ee13889399ca71a026549a2 "hello"))
+        let spend = CoinSpend::new(test_coin, Program::new(vec![1_u8].into()), solution.into());
+        let msg = b"goodbye"; // bad message
+        let mut result = msg.to_vec();
+        result.extend(
+            [
+                test_coin.parent_coin_info.as_slice(),
+                test_coin.puzzle_hash.as_slice(),
+                TEST_CONSTANTS
+                    .agg_sig_parent_puzzle_additional_data
+                    .as_slice(),
+            ]
+            .concat(),
+        );
+        let sig = sign(&sk, result.as_slice());
+        let coin_spends: Vec<CoinSpend> = vec![spend];
+        let spend_bundle = SpendBundle {
+            coin_spends,
+            aggregated_signature: sig,
+        };
+        let result = validate_clvm_and_signature(
+            &spend_bundle,
+            TEST_CONSTANTS.max_block_cost_clvm,
+            &TEST_CONSTANTS,
+            246,
+        );
+        assert!(matches!(result, Err(ErrorCode::BadAggregateSignature)));
```

Comment:
the only difference between this and the other "test wrong message" block is the condition code. It seems like a perfect opportunity for parameterizing the test with `rstest`